planet/http.py

```diff
@@ -431,10 +431,10 @@ def client(self,
 
         """
         # To avoid circular dependency.
-        from planet.clients import client_directory
+        from planet.clients import _client_directory
 
         try:
-            return client_directory[name](self, base_url=base_url)
+            return _client_directory[name](self, base_url=base_url)
```

Comment:
This way the Session class only needs to know that it can get a client with a name, it's not responsible for the directory. Looser coupling.